diff --git a/src/types.js b/src/types.js
index 7c7c2b93094a1bced2b9dc387edc5c403e2c5864..818b6893f1c68932c8dcb448be961e9ca10c6d2e 100644
--- a/src/types.js
+++ b/src/types.js
@@ -317,15 +317,20 @@ function arrayParserLoop(s, x, parser, typarray) {
 
 export const toCamel = x => {
   let str = x[0]
-  for (let i = 1; i < x.length; i++)
-    str += x[i] === '_' ? x[++i].toUpperCase() : x[i]
+  for (let i = 1; i < x.length; i++) {
+    if (x[i] === '_' && i + 1 < x.length) {
+      str += x[++i].toUpperCase()
+    } else if (x[i] !== '_') {
+      str += x[i]
+    }   
+  }
   return str
 }
 
 export const toPascal = x => {
   let str = x[0].toUpperCase()
   for (let i = 1; i < x.length; i++)
-    str += x[i] === '_' ? x[++i].toUpperCase() : x[i]
+    str += x[i] === '_' ? x[++i]?.toUpperCase() : x[i]
   return str
 }
 
